temporian/implementation/numpy/data/event.py

```diff
@@ -89,6 +105,165 @@ def schema(self) -> Event:
             sampling=self.sampling.names,
         )
 
+    @staticmethod
+    def from_dataframe(
+        df: pd.DataFrame,
+        index_names: List[str],
```

Comment:
Having the indexing logic is great. However, would it make sense to move this logic to the "addIndex" (or something similar) operator instead?

To speed-up the work, you can submit this PR as it is, and maybe make the move in another PR.